### ISSUE TITLE
Accordion Element gives NullReferenceException when saved as a prefab

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#541]
<sup>Issue created by J.R. Mitchell as [Bitbucket Issue #&#8203;364](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364) on 2021.01.25 03:57.</sup>
When an accordion element is saved as a prefab without the accordion group, m\_Accordion ends up as null and it gives a NullReferenceException in several circumstances

UnityEngine.UI.Extensions.AccordionElement.OnValidate \(\) \(at Library/PackageCache/com.unity.uiextensions@2.2.4/Runtime/Scripts/Controls/Accordion/AccordionElement.cs:70\) \(On scene load, play, and when inspector is changed\)

UnityEngine.UI.Extensions.AccordionElement.OnValueChanged \(System.Boolean state\) \(at Library/PackageCache/com.unity.uiextensions@2.2.4/Runtime/Scripts/Controls/Accordion/AccordionElement.cs:106\) \(When isOn is toggled\)

Possibly related - when opening the accordion element prefab for editing, it is stuck as collapsed. I’m unsure if this is caused by the script or how my prefab is arranged.

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.01.26 10:09](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-59817515), @SimonDarksideJ commented:</sup>
Interesting @{557058:a4d12867-f51b-4edc-8e19-f7da90498887} , can you detail the use case you are trying to achieve?  Just so I can start testing / investigating.

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.01.26 16:40](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-59820835), J.R. Mitchell commented:</sup>
I am making an editor for an XML-formatted file type that uses the AccordionElement for each node. It is saved as a prefab and added to the scene as needed.

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.03.07 01:05](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-60107678), J.R. Mitchell commented:</sup>
Any updates on this?  For now I’m just adding a null check on line 66 every time I open Unity since it overwrites my change.

66:

```csharp
if (le != null && m_Accordion != null)
```

102:

```csharp
if (transition == Accordion.Transition.Instant && m_Accordion != null)
```

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.03.10 15:24](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-60132552), @SimonDarksideJ commented:</sup>
Thanks for the update / patch, will review getting that in and updated in to the UPM package.

Alternatively, you could download the asset to make the change permanent for now.

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 15:09](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-60494004), @SimonDarksideJ commented:</sup>
Sorry for the delay in getting to this, I’ve pulled in your changes and they will be in the 2.2.5 release \(hopefully getting out today\) @{557058:a4d12867-f51b-4edc-8e19-f7da90498887}

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 15:12](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-60494022), @SimonDarksideJ commented:</sup>
Applied J.R. Mitchell's fix for the Accordion Controls/Accordion/AccordionElement.cs

resolves: #364


Referenced by a33a8467f508

**Comment by SimonDarksideJ** [comment]
<sup>On 2021.05.10 15:12 @SimonDarksideJ modified issue:</sup>
**status** changed `new` → `resolved`

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 15:14](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-60494049), @SimonDarksideJ commented:</sup>
Resolved in 2.2.5

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 18:38](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/364#comment-60495267), @SimonDarksideJ commented:</sup>
Merged in development (pull request #122)

Development

* Merged in updated_new_input_system_check (pull request #117)

Updated the new Input system switch and tested against 2021

* Updated LineRenderer to initialise with a 1 point array, this resolves an issue with Unity putting the previous controls vertext array in an uninitialised control.
Also updated examples checkout

* Merged in UpdatedLineRender (pull request #118)

UILineRender now initilises with a point of 1 to correct Unity issue

* Applied J.R. Mitchell's fix for the Accordion Controls/Accordion/AccordionElement.cs

resolves: #364

* Merged in AccordionFIx (pull request #119)

Update to Accordion to allow for prefab Accordion Elements

* Resolved issue where the Content Scroll snap would cause issue with only 1 child.
Also tested and fixed working with no children.

Resolves #362

* Merged in content_scroll_snap_fix (pull request #120)

ContentScrollSnap update

* Updated the PaginationManager to override if the ScrollSnap is in motion.
BuyerBeware, Unity's toggles can get in to a state where they are clicked but not clicked.

* Merged in PaginationFix (pull request #121)

Updated the PaginationManager to override if the ScrollSnap is in motion.

* Updated release notes and bumped version to release


Referenced by 383e19fb0512
